firewood/src/merkle.rs

```diff
@@ -554,7 +560,7 @@ impl<S: ReadableStorage> Merkle<NodeStore<MutableProposal, S>> {
                 match node {
                     Node::Branch(ref mut branch) => {
                         #[expect(clippy::indexing_slicing)]
-                        let child = match std::mem::take(&mut branch.children[child_index as usize])
+                        let child: Node<Option<Child>> = match std::mem::take(&mut branch.children[child_index as usize]).as_child_option()
```

Comment:
You probably want a `take` method on T:

```suggestion
                        let child: Node<Option<Child>> = match &mut branch.children[child_index as usize].take()
```

The implementation is something like:

```
fn take(&smut self) -> Option<Child> {
    std::mem::take(self)
}
```

This is probably doable with a lock too, but I think you'll have trouble because you probably need to keep holding the lock after the take.